### PR TITLE
Update C-Chain config docs

### PR DESCRIPTION
**Pull request by aaronbuchwald**
This PR updates the C-Chain config docs to reflect the latest configuration options.

### CODE REVIEW

File: docs/build/references/command-line-interface.md [hunk 1]
```diff
@@ -309,28 +309,59 @@ The default C-Chain config is:
...
-  "tx-pool-api-enabled": false,
-  "debug-api-enabled": false,
-  "web3-api-enabled": true,
-  "local-txs-enabled": false,
-  "pruning-enabled": false,
-  "api-max-duration": 0, // Default to no maximum
-  "api-max-blocks-per-request": 0, // Default to no maximum
-  "allow-unfinalized-queries": false,
-  "log-level": "info"
+   "snowman-api-enabled":false,
```
Comment (dhrubabasu) [inline docs/build/references/command-line-interface.md:312]: nit: let's keep it to an indent-size of two spaces and add a space between the key and value

File: docs/build/references/command-line-interface.md [hunk 2]
```diff
@@ -379,12 +414,74 @@ The maximum gas to be consumed by an RPC Call \(used in `eth_estimateGas` and `e
 
 Global transaction fee \(price \* gaslimit\) cap \(measured in AVAX\) for send-transction variants. Defaults to 100.
```
Comment (dhrubabasu) [inline docs/build/references/command-line-interface.md:415]:
```suggestion
Global transaction fee \(price \* gaslimit\) cap \(measured in AVAX\) for send-transaction variants. Defaults to 100.
```

File: docs/build/references/command-line-interface.md [hunk 3]
```diff
@@ -379,12 +414,74 @@ The maximum gas to be consumed by an RPC Call \(used in `eth_estimateGas` and `e
...
+
+**Transaction Pool**
+
+`local-txs-enabled` \(boolean\):
+
+Enables local transaction handling (prioritizes transactions submitted through this node). Defaults to false.
+
+`allow-unprotected-txs` \(boolean\):
+
+If true, the APIs will allow transactions that are not replay protected to be issued through this node. Defaults to false.
```
Comment (dhrubabasu) [inline docs/build/references/command-line-interface.md:441]:
```suggestion
If true, the APIs will allow transactions that are not replay protected (EIP-155) to be issued through this node. Defaults to false.
```

File: docs/build/references/command-line-interface.md [hunk 4]
```diff
@@ -379,12 +414,74 @@ The maximum gas to be consumed by an RPC Call \(used in `eth_estimateGas` and `e
...
+
+`metrics-enabled` \(boolean\):
+
+Enables metrics. Defaults to false.
+
+`metrics-expensive-enabled` \(boolean\):
+
+Enables expensive metrics. Defaults to false.
+
+**Databaset**
```
Comment (dhrubabasu) [inline docs/build/references/command-line-interface.md:465]:
```suggestion
**Database**
```

**Review by dhrubabasu — CHANGES_REQUESTED** [review]
Left a few nits

**Review by dhrubabasu — APPROVED** [review]

**Review by yulin-dong — APPROVED** [review]

**Pull request merged**